File: pytools/data/data.code.golang.client/ack_scene_move.go
package proto

import (
	"packet"
)

type AckSceneMove struct {
	sceneRotPos              *MsgSceneRotPos
	forward                  *MsgSceneVector3
	aniName                  string
	xAxis                    int16
	uid                      uint32
}

func AckSceneMoveDecode(pack *packet.Packet) *AckSceneMove {
	ackSceneMove := &AckSceneMove{}

	ackSceneMove.sceneRotPos = MsgSceneRotPosDecode(pack)
	ackSceneMove.forward = MsgSceneVector3Decode(pack)
	ackSceneMove.aniName = pack.ReadString()
	ackSceneMove.xAxis = pack.ReadInt16()
	ackSceneMove.uid = pack.ReadUint32()
	return ackSceneMove
}

func (this *AckSceneMove) GetSceneRotPos() *MsgSceneRotPos {
	return this.sceneRotPos
}

func (this *AckSceneMove) GetForward() *MsgSceneVector3 {
	return this.forward
}

func (this *AckSceneMove) GetAniName() string {
	return this.aniName
}

func (this *AckSceneMove) GetXAxis() int16 {
	return this.xAxis
}

func (this *AckSceneMove) GetUid() uint32 {
	return this.uid
}

File: pytools/examples/php/client/protocol/AckSceneMove.php
<?php
namespace protocol;

use network\Packet;


class AckSceneMove
{
	private $sceneRotPos = null;
	private $forward = null;
	private $aniName = '';
	private $xAxis = 0;
	private $uid = 0;


	public function __construct(&$packet)
	{
		$this->decode($packet);
	}

	public function decode(&$packet)
	{
		$this->sceneRotPos = new MsgSceneRotPos($packet);
		$this->forward = new MsgSceneVector3($packet);
		$this->aniName = $packet->readString();
		$this->xAxis = $packet->readI16();
		$this->uid = $packet->readU32();
	}


	public function getSceneRotPos()
	{
		return $this->sceneRotPos;
	}

	public function getForward()
	{
		return $this->forward;
	}

	public function getAniName()
	{
		return $this->aniName;
	}

	public function getXAxis()
	{
		return $this->xAxis;
	}

	public function getUid()
	{
		return $this->uid;
	}

}
